### PR TITLE
fix: deprecation warnings regarding uniform type identifiers - WPB-8778

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8778" title="WPB-8778" target="_blank">TaskWPB-8778</a>  Clean up code base
  </td></table>
  <br />
 

### Issue

In Swift uniform type identifiers are a dedicated type instead of a string.
Therefore Apple deprecated various API methods and types.

This PR migrates the affected code.

### Testing

No changes should be noticed on the user side, especially regarding file previews (pdf, image, video, etc.)

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [x] Make sure you use the API for UI elements that support large fonts.
- [x] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [x] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: wire-ios/WireCommonComponents/FilePreviewGenerator/AggregateFilePreviewGenerator.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+
+struct AggregateFilePreviewGenerator: FilePreviewGenerator {
+
+    let generators: [FilePreviewGenerator]
+    let thumbnailSize: CGSize
+    let callbackQueue: OperationQueue
```
Comment (caldrian) [inline wire-ios/WireCommonComponents/FilePreviewGenerator/AggregateFilePreviewGenerator.swift:25]: Unused property to be removed.

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/AudioEffectsPickerViewController.swift [hunk 2]
```diff
@@ -188,13 +188,14 @@ final class AudioEffectsPickerViewController: UIViewController {
 
     private func loadLevels() {
         let url = URL(fileURLWithPath: recordingPath)
-        fileMetadataGenerator.metadataForFileAtURL(url, UTI: url.UTI(), name: url.lastPathComponent) { metadata in
-            DispatchQueue.main.async(execute: {
+        Task {
+            let metadata = await fileMetadataGenerator.metadataForFile(at: url)
+            await MainActor.run {
```
Comment (jullianm) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/AudioEffectsPickerViewController.swift]: Could we do `Task { @MainActor  in {...} } here to be aligned with what you did right after (line 83 in ConversationInputBarViewController+Files) ?

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/Debug Report/SettingsDebugReportViewController.swift [hunk 3]
```diff
@@ -123,11 +123,11 @@ class SettingsDebugReportViewController: UIViewController {
     // MARK: - Actions
 
     @objc private func didTapSendReport(sender: UIView) {
-        viewModel.sendReport(sender: sender)
+        Task { viewModel.sendReport(sender: sender) }
```
Comment (jullianm) [inline wire-ios/Wire-iOS/Sources/UserInterface/Settings/Debug Report/SettingsDebugReportViewController.swift]: Do we need a Task here ? I don't see any awaiting code

File: wire-ios/WireCommonComponents/FilePreviewGenerator/MovieFilePreviewGenerator.swift [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+struct MovieFilePreviewGenerator: FilePreviewGenerator {
+
+    let thumbnailSize: CGSize
+
+    func supportsPreviewGenerationForFile(at url: URL) -> Bool {
+        guard let uniformType = url.uniformType else { return false }
+        return AVURLAsset.wr_isAudioVisualUniformType(uniformType)
+    }
+
+    func generatePreviewForFile(at url: URL) async throws -> UIImage {
```
Comment (jullianm) [inline wire-ios/WireCommonComponents/FilePreviewGenerator/MovieFilePreviewGenerator.swift]: Does this method need to be marked as async ?

File: wire-ios/WireCommonComponents/FilePreviewGenerator/PDFFilePreviewGenerator.swift [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+struct PDFFilePreviewGenerator: FilePreviewGenerator {
+
+    let thumbnailSize: CGSize
+
+    func supportsPreviewGenerationForFile(at url: URL) -> Bool {
+        url.uniformType?.conforms(to: .pdf) ?? false
+    }
+
+    func generatePreviewForFile(at url: URL) async throws -> UIImage {
```
Comment (jullianm) [inline wire-ios/WireCommonComponents/FilePreviewGenerator/PDFFilePreviewGenerator.swift]: Does it need to be async ?

**Review by jullianm — APPROVED** [review]
LGTM (just some minor comments)

File: wire-ios/Wire-iOS Tests/ReferenceImages/AuthenticationInterfaceBuilderTests/testActivationScreen_Email.1.png
Comment (samwyndham) [inline wire-ios/Wire-iOS Tests/ReferenceImages/AuthenticationInterfaceBuilderTests/testActivationScreen_Email.1.png:1]: Why the changes to this and other screenshot tests?

File: wire-ios/Wire-iOS Share Extension/UnsentSendable.swift [hunk 6]
```diff
@@ -278,19 +278,14 @@ class UnsentFileSendable: UnsentSendableBase, UnsentSendable {
 
                 // Generating PDF previews can be expensive. To avoid hitting the Share Extension's
                 // memory budget we should avoid to generate previews if the file is a PDF.
-                guard let type = UTType(url.UTI()), type.conforms(to: UTType.pdf) else {
+                guard let uniformType = url.uniformType, uniformType.conforms(to: .pdf) else {
                     self?.metadata = ZMFileMetadata(fileURL: url)
-                    completion()
-                    return
+                    return completion()
```
Comment (samwyndham) [inline wire-ios/Wire-iOS Share Extension/UnsentSendable.swift:283]: This works because `completion` returns void right? I'm not sure how I feel about this style. When I look at it I assume `completion`  will return something non void and that the method that it is enclosed in also returns something non void. What are the benefits?

File: wire-ios/Wire-iOS Tests/FilePreviewGeneratorTests.swift [hunk 7]
```diff
@@ -23,31 +23,36 @@ import XCTest
...
+    func testThatItThrowsOnHugePDF() async throws {
+
+        // Given
+        let pdfURL = try XCTUnwrap(bundle.url(forResource: "huge", withExtension: "pdf"))
+        let sut = PDFFilePreviewGenerator(thumbnailSize: CGSize(width: 100, height: 100))
+
+        // When
+        do {
+            _ = try await sut.generatePreviewForFile(at: pdfURL)
+            XCTFail("Unexpected success")
```
Comment (samwyndham) [inline wire-ios/Wire-iOS Tests/FilePreviewGeneratorTests.swift:53]: Out of scope but we should probably create a custom matcher to handle these expecting to throw await situations

File: wire-ios/Wire-iOS Tests/ReferenceImages/AuthenticationInterfaceBuilderTests/testActivationScreen_Email.1.png
Comment (caldrian) [reply wire-ios/Wire-iOS Tests/ReferenceImages/AuthenticationInterfaceBuilderTests/testActivationScreen_Email.1.png:1]: I don't know how these changes are related to my PR.
It looks like the content of the view has shifted a bit to the left.

File: wire-ios/WireCommonComponents/FilePreviewGenerator/PDFFilePreviewGenerator.swift (hunk 5 above)
Comment (caldrian) [reply wire-ios/WireCommonComponents/FilePreviewGenerator/PDFFilePreviewGenerator.swift]: It doesn't, fixed, thank you.

File: wire-ios/WireCommonComponents/FilePreviewGenerator/MovieFilePreviewGenerator.swift (hunk 4 above)
Comment (caldrian) [reply wire-ios/WireCommonComponents/FilePreviewGenerator/MovieFilePreviewGenerator.swift]: It doesn't, fixed, thank you.

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/Debug Report/SettingsDebugReportViewController.swift (hunk 3 above)
Comment (caldrian) [reply wire-ios/Wire-iOS/Sources/UserInterface/Settings/Debug Report/SettingsDebugReportViewController.swift]: Fixed, thanks!

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/AudioEffectsPickerViewController.swift (hunk 2 above)
Comment (caldrian) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/AudioEffectsPickerViewController.swift]: Fixed, thanks.

**Review by samwyndham — COMMENTED** [review]
Nice work! I think it would be good to understand the reason for the snapshot test failures before merging though - and ideally removing the snapshot updates. I'm just concerned there is a subtle change we are missing.

File: wire-ios/Wire-iOS Tests/FilePreviewGeneratorTests.swift (hunk 7 above)
Comment (caldrian) [reply wire-ios/Wire-iOS Tests/FilePreviewGeneratorTests.swift:53]: I'm not sure I can follow you regarding this point, but generally I'm not a fan of little helpers and custom overrides of the SDK API. It creates confusion and makes the code less readable.

One example: the file Sequence+Async.swift
https://github.com/wireapp/wire-ios/pull/1284/files#diff-e04d4a1ce91dfd894ba34403a7d3106577fa4ced1180e137751208db506526ccL19

Maybe we're talking about different things.

File: wire-ios/Wire-iOS Share Extension/UnsentSendable.swift (hunk 6 above)
Comment (caldrian) [reply wire-ios/Wire-iOS Share Extension/UnsentSendable.swift:283]: Just that the code is a little more compact. I'll revert it.

File: wire-ios/Wire-iOS Tests/FilePreviewGeneratorTests.swift (hunk 7 above)
Comment (samwyndham) [reply wire-ios/Wire-iOS Tests/FilePreviewGeneratorTests.swift:53]: Sorry I'll clarify. I mean ideally we could test this using `XCTAssertThrowsError` but that doesn't work for `async` so it would be helpful to have a version of `XCTAssertThrowsError` that also works for async.

**Review by samwyndham — APPROVED** [review]

File: wire-ios/Wire-iOS Tests/FilePreviewGeneratorTests.swift (hunk 7 above)
Comment (samwyndham) [reply wire-ios/Wire-iOS Tests/FilePreviewGeneratorTests.swift:53]: @caldrian Feel free to close this comment. I wasn't asking for a change, just thinking out loud.

**Review by agisilaos — APPROVED** [review]
